src/webaudio-mediasource.js

```diff
@@ -0,0 +1,459 @@
+import WebAudio from './webaudio';
+import * as util from './util';
+
+/**
+ * MediaElement backend
+ */
+export default class WebAudioMedia extends WebAudio {
+    /** @private */
+    static scriptBufferSize = 256;
+    /** @private */
+    audioContext = null;
+    /** @private */
+    offlineAudioContext = null;
+    /**
+     * Construct the backend
+     *
+     * @param {WavesurferParams} params Wavesurfer parameters
+     */
+    constructor(params) {
+        super(params);
+        /** @private */
+        this.params = params;
+        /** @private */
+        this.params = params;
+        /** @private */
+        this.ac =
+            params.audioContext ||
+            (this.supportsWebAudio() ? this.getAudioContext() : {});
+
+        // Dummy media to catch errors
+        /** @private */
+        this.media = {
+            currentTime: 0,
+            duration: 0,
+            paused: true,
+            playbackRate: 1,
+            play() {},
+            pause() {},
+            volume: 0
+        };
+
+        /** @private */
+        this.mediaType = params.mediaType.toLowerCase();
+        /** @private */
+        this.elementPosition = params.elementPosition;
+        /** @private */
+        this.peaks = null;
+        /** @private */
+        this.playbackRate = 1;
+        /** @private */
+        this.volume = 1;
+        /** @private */
+        this.isMuted = false;
+        /** @private */
+        this.buffer = null;
+        /** @private */
+        this.onPlayEnd = null;
+        /** @private */
+        this.sourceMediaElement = null;
+        /** @private */
+        this.scriptNode = null;
+        /** @private */
+        this.gainNode = null;
+        /** @private */
+        this.analyser = null;
+    }
+
+    /**
+     * Initialise the backend, called in `wavesurfer.createBackend()`
+     */
+    init() {
+        this.setPlaybackRate(this.params.audioRate);
+        this.createTimer();
+        this.createVolumeNode();
+        this.createScriptNode();
+        this.createAnalyserNode();
+    }
+    /**
+     * Create a timer to provide a more precise `audioprocess` event.
+     *
+     * @private
+     */
+    createTimer() {
+        const onAudioProcess = () => {
+            if (this.isPaused()) {
+                return;
+            }
+            this.fireEvent('audioprocess', this.getCurrentTime());
+
+            // Call again in the next frame
+            util.frame(onAudioProcess)();
+        };
+
+        this.on('play', onAudioProcess);
+
+        // Update the progress one more time to prevent it from being stuck in
+        // case of lower framerates
+        this.on('pause', () => {
+            this.fireEvent('audioprocess', this.getCurrentTime());
+        });
+    }
+    /** @private */
+    createScriptNode() {
```

Comment:
Isn't this method exactly the same in the parent WebAudio class?

Reply:
Yes, i've added only createSource method which allow me to create MediaElementSourceNode (instead of AudioBufferSourceNode) and handle audio resource with API Web Audio. I've duplicated some code, only to show you my solution in a different file,  to give more clarity to my purpose.

Reply:
i worry about if we need a new backend to handle this, i need time to check it